app/Http/Requests/StoreSettingRequest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class StoreSettingRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'key' => ['required', 'string', 'min:2', 'max:255', Rule::unique('settings')->where(function ($query)
+            {
+                return $query->where('user_id', auth()->user()->id);
+            })],
+            'value' => ['required', 'string', 'min:2', 'max:255']
+        ];
+    }
+}
```

Comment:
This looks good! When you have a closure rule in the rule list You can separate the rules into their own lines to improve readability like this:

```php
return [
      'key' => [
      'required', 
      'string', 
      'min:2', 
      'max:255', 
      Rule::unique('settings')->where(function ($query)
      {
           return $query->where('user_id', auth()->user()->id);
      })],
      'value' => ['required', 'string', 'min:2', 'max:255']
  ];
```